content/chapters/data/lab/content/memory-support.md

```diff
@@ -0,0 +1,85 @@
+## Memory Support
+
+**Manual memory management** (MMM) is one of the most difficult tasks.
+Even experienced programmers make mistakes when tackling such a complicated endeavor.
+As a consequence, the programming world has been migrating towards languages that offer automatic memory management (AMM).
+AMM programming languages typically offer a garbage collector that tracks down the usage of objects and frees memory once no references exist to a given object.
+As a consequence garbage collected programming languages are easier to use and safer.
+However, this comes with a cost: the garbage collector, in most cases, requires a significant amount of resources to run.
+Therefore, for performance-critical systems, MMM is still the preferred solution.
+
+A middle-ground between programming languages that have AMM (Java, Python, Rust, D) and those that do not (C, C++) is represented by those languages that do not have builtin AMM but offer the possibility to implement it as a library solution (C++, D).
```

Comment:
```suggestion
A middle-ground between programming languages that have AMM (Java, Python, D) and those that do not (C, C++, Rust) is represented by those languages that do not have builtin AMM but offer the possibility to implement it as a library solution (C++, D, Rust).
```

Rust does not have AMM [1]. But it has a reference counting structure in the std library [2].

[1] https://stackoverflow.com/questions/40658045/does-rusts-memory-management-result-in-fragmented-memory#:~:text=Rust%20does%20not%20have%20automatic,going%20on%20behind%20the%20scenes.
[2] https://doc.rust-lang.org/stable/std/rc/struct.Rc.html

Reply:
Sorry, i had Swift in mind, but typed Rust ...